rskj-core/src/main/java/co/rsk/core/bc/BlockExecutor.java

```diff
@@ -195,16 +195,16 @@ public boolean validate(Block block, BlockResult result) {
         return true;
     }
 
-    private boolean validateStateRoot(BlockHeader header, BlockResult result) {
+    @VisibleForTesting
+    boolean validateStateRoot(BlockHeader header, BlockResult result) {
         boolean isRskip85Enabled = activationConfig.isActive(RSKIP85, header.getNumber());
         if (!isRskip85Enabled) {
             return true;
         }
 
         boolean isRskip126Enabled = activationConfig.isActive(RSKIP126, header.getNumber());
         if (!isRskip126Enabled) {
-            byte[] orchidStateRoot = stateRootHandler.convert(header, result.getFinalState()).getBytes();
-            return Arrays.equals(orchidStateRoot, header.getStateRoot());
+            return true;
         }
 
         // we only validate state roots of blocks newer than 0.5.0 activation
```

Comment:
This comments seems to be outdated now :)

Reply:
yeah, it's confusing a bit) I've updated that msg a bit. So the point here is that we do not validate roots before `Rskip 126`, but after its activation we do.